api/guids/views.py

```diff
@@ -37,7 +37,10 @@ def get_serializer_class(self):
 
     def get(self, request, **kwargs):
         url = self.get_redirect_url(**kwargs)
+        view_only_token = self.request.query_params.get('view_only')
         if url:
+            if view_only_token:
+                url += '?view_only={}'.format(view_only_token)
```

Comment:
Is it possible for there to be query params on `url` before this point? If that's the case, then the `view_only` param will need to be appended with a `&`. You can use `furl` to take care of this if you want.